app/Main.hs

```diff
@@ -1,4 +1,77 @@
+{-# LANGUAGE OverloadedStrings #-}
+
 module Main where
 
+import Control.Monad (forever)
+import Data.Conduit (runConduit, (.|))
+import Data.List (singleton)
+import Kafka.Producer
+  ( ProducerProperties (..),
+    Timeout (..),
+    brokersList,
+    sendTimeout,
+  )
+import Options.Applicative
+  ( Parser,
+    auto,
+    command,
+    execParser,
+    fullDesc,
+    helper,
+    hsubparser,
+    idm,
+    info,
+    long,
+    metavar,
+    option,
+    progDesc,
+    short,
+    str,
+    value,
+    (<**>),
+  )
+import SatSim.Gen.Producer (kafkaBatchProducer, timeProducer)
+import SatSim.Quantities (Seconds (..))
+
+data Command
+  = ProduceEvery ProducerProperties Int Seconds
+  | RunScheduler
+
+producerPropertiesParser :: Parser ProducerProperties
+producerPropertiesParser =
+  let brokerAddress = option str (long "broker-address" <> metavar "BROKER_PORT" <> value "localhost:9092")
```

Comment:
TIL -- `str` is `IsString a => ReadM a`, while `auto` is `Read a => ReadM a`. If you want to parse a string argument, then the `Read String` instance will require quoting the argument when you pass it --

```haskell
ghci> (read "\"abcd\"") :: String
"abcd"
ghci> (read "abcd") :: String
"*** Exception: Prelude.read: no parse
```